crates/astria-composer/src/composer.rs

```diff
@@ -15,42 +28,113 @@ use crate::{
         self,
         ApiServer,
     },
-    searcher::Searcher,
+    composer_status::ComposerStatus,
+    executor,
+    executor::{
+        Executor,
+        ExecutorHandle,
+    },
+    geth_collector,
+    geth_collector::GethCollector,
+    grpc_collector::GrpcCollector,
+    rollup::Rollup,
     Config,
 };
 
-/// `Composer` is a service responsible for submitting transactions to the Astria
-/// Shared Sequencer.
+/// `Composer` is a service responsible for spinning up `GethCollectors` which are responsible
+/// for fetching pending transactions submitted to the rollup Geth nodes and then passing them
+/// downstream for the executor to process. Thus, a composer can have multiple collectors running
+/// at the same time funneling data from multiple rollup nodes.
 pub struct Composer {
     /// `ApiServer` is used for monitoring status of the Composer service.
     api_server: ApiServer,
-    /// `Searcher` establishes connections to individual rollup nodes, receiving
-    /// pending transactions from them and wraps them as sequencer transactions
-    /// for submission.
-    searcher: Searcher,
+    /// `ComposerStatusSender` is used to announce the current status of the Composer for other
+    /// modules in the crate to use.
+    composer_status_sender: watch::Sender<ComposerStatus>,
+    /// `ExecutorHandle` to communicate SequenceActions to the Executor
+    /// This is at the Composer level to allow its sharing to various different collectors.
+    executor_handle: ExecutorHandle,
+    /// `Executor` is responsible for signing and submitting sequencer transactions
+    /// The sequencer transactions are received from various collectors.
+    executor: Executor,
+    /// `GethCollectors` is the collection of geth collectors and their rollup names.
+    geth_collectors: HashMap<String, GethCollector>,
+    /// `GethCollectorStatuses` The collection of the geth collector statuses.
+    geth_collector_statuses: HashMap<String, watch::Receiver<geth_collector::Status>>,
+    /// `GethCollectorTasks` is the set of tasks tracking if the geth collectors are still running.
+    geth_collector_tasks: JoinMap<String, eyre::Result<()>>,
+    /// `Rollups` The map of chain ID to the URLs to which geth collectors should connect.
+    rollups: HashMap<String, String>,
```

Comment:
```suggestion
    rollup_id_to_url: HashMap<String, String>,
```

more descriptive is good here